src/libsyntax/feature_gate.rs

```diff
@@ -230,24 +250,23 @@ impl<'a, 'v> Visitor<'v> for MacroVisitor<'a> {
         let ast::MacInvocTT(ref path, _, _) = mac.node;
         let id = path.segments.last().unwrap().identifier;
 
+        // Issue 22234: If you add a new case here, make sure to also
+        // add code to catch the macro during or after expansion.
+
         if id == token::str_to_ident("asm") {
```

Comment:
Theoretically these are redundant now (that is, they'd be caught by expansion), but I guess you have a motivation for keeping them around?